### PR TITLE
 check if file exists before opening it

### LINKED ISSUE #164: Check the existence of the file before opening it
In the list of completed downloads you must check if the file exists before opening it. If it does not exist, please show a message that the file does not exist or has been deleted, something like that.

### PR DESCRIPTION
**Pull request by xibr**
check if file exists before opening it, this PR fixes #164 

## Pre-launch Checklist

- [x] I read the [Contributor Guide](CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- If you made changes to the code:
  - [ ] I added new tests to check the change I am making or feature I am adding.
  - [ ] All existing and new tests are passing.

### CODE REVIEW

File: app/src/main/java/com/tachibana/downloader/core/system/FileSystemFacadeImpl.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
 
 import com.tachibana.downloader.core.exception.FileAlreadyExistsException;
 
+import org.jetbrains.annotations.NotNull;
```
Comment (proninyaroslav) [inline app/src/main/java/com/tachibana/downloader/core/system/FileSystemFacadeImpl.java:34]: Please remove this import

File: app/src/main/java/com/tachibana/downloader/core/system/SafFsModule.java [hunk 2]
```diff
@@ -31,6 +31,8 @@
 
 import androidx.annotation.NonNull;
 
+import org.jetbrains.annotations.NotNull;
```
Comment (proninyaroslav) [inline app/src/main/java/com/tachibana/downloader/core/system/SafFsModule.java:34]: And also please remove this import

File: app/src/main/java/com/tachibana/downloader/ui/main/FinishedDownloadsFragment.java [hunk 3]
```diff
@@ -123,11 +124,13 @@ public void onSaveInstanceState(@NonNull Bundle outState)
...
-    {
-        startActivity(Intent.createChooser(
-                Utils.createOpenFileIntent(activity.getApplicationContext(), item.info),
-                getString(R.string.open_using)));
+    public void onItemClicked(@NonNull DownloadItem item) {
+        Intent file = Utils.createOpenFileIntent(activity.getApplicationContext(), item.info);
+        if (file != null) {
+            startActivity(Intent.createChooser(file, getString(R.string.open_using)));
+        } else {
+            Toast.makeText(activity.getApplicationContext(), "file not available!", Toast.LENGTH_SHORT).show();
```
Comment (proninyaroslav) [inline app/src/main/java/com/tachibana/downloader/ui/main/FinishedDownloadsFragment.java:132]: Add a new string to `strings.xml` to localize this message.

**Review by proninyaroslav — CHANGES_REQUESTED** [review]

**Review by proninyaroslav — APPROVED** [review]

File: app/src/main/java/com/tachibana/downloader/ui/main/FinishedDownloadsFragment.java (hunk 3 above)
Comment (xibr) [reply app/src/main/java/com/tachibana/downloader/ui/main/FinishedDownloadsFragment.java]: Done.

**Comment by proninyaroslav** [comment]
Thank you!

**Pull request merged**